src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -214,8 +214,11 @@ public static List<NodeInfo> getXpathItems(String xpath, AbstractNode rootNode)
         final XPathDynamicContext xpathDynamicContext = xpathExpression
                 .createDynamicContext(rootNode);
         final List<Item> items = xpathExpression.evaluate(xpathDynamicContext);
-        return items.stream()
-                .map(NodeInfo.class::cast)
-                .collect(Collectors.toUnmodifiableList());
+        final List<NodeInfo> nodeInfoList = new ArrayList<>();
```

Comment:
please make method `UnmodifiableCollectionUtil.unmodifiableList(items, NodeInfo.class)`
to keep what pitest naturaly see as not required code is special place/class.